### PR TITLE
Add viewContainer contribution point

### LINKED ISSUE #200880: Debug viewContainer menu contribution points
Apologies if this is too similar to existing requests, but I'm keen to pull together a common request for adding debug contribution points.

Similar requests:

- https://github.com/microsoft/vscode/issues/150161
- https://github.com/microsoft/vscode/issues/129536
- https://github.com/microsoft/vscode/issues/171548

And related:

- https://github.com/microsoft/vscode/issues/197287

Rationale:

Debugging often needs deep integration between views (e.g. embedded debugging) where they are all driven from a common context and can interact with other. A common way to add these links/commands is through contribution points and some of the provided debug views lack these extension points. Can I suggest contribution points in context menus for the following:

- Watch view  (e.g. to open the value in a memory viewer)
- Breakpoints view (e.g. to expose a 'run to breakpoint' command) 
- Disassembly view (e.g. to show source, manage breakpoints, show memory)
- Debug viewlet bar (e.g. to add a command to launch a custom GUI to create a complex debug configuration)

I assume this is the debug viewlet:

Screenshot 2023-12-14 at 19 03 34

Happy to work on a PR if this is of interest.

### PR DESCRIPTION
**Pull request by thegecko**
As discussed in #200880 we are keen to add a contribution point to the debug viewlet.

This proposal implements this as well as allowing contributions to other view containers.

To use this proposal, the `package.json` would look similar to:

```json
{
        ...
	"enabledApiProposals": [
		"viewContainerContext"
	],
	"contributes": {
		"commands": [
			{
				"command": "proposed.dummyMenu",
				"title": "Dummy Menu",
				"icon": "$(beaker)"
			}
		],
		"menus": {
			"viewContainer/title": [
				{
					"command": "proposed.dummyMenu",
					"group": "navigation",
					"when": "viewContainer == workbench.view.debug"
				}
			]
		}
	},
        ...
```

An example can be found here: https://github.com/thegecko/vscode-proposed-debug

Screenshot 2024-05-11 at 12 08 52

Partially fixes: #200880

### CODE REVIEW

**Comment by jrieken** [comment]
It this be `viewContainer/...` instead of `view/container/..`?

File: src/vs/workbench/services/extensions/common/extensionsApiProposals.ts [hunk 1]
```diff
@@ -127,6 +127,7 @@ export const allApiProposals = Object.freeze({
 	treeViewReveal: 'https://raw.githubusercontent.com/microsoft/vscode/main/src/vscode-dts/vscode.proposed.treeViewReveal.d.ts',
 	tunnelFactory: 'https://raw.githubusercontent.com/microsoft/vscode/main/src/vscode-dts/vscode.proposed.tunnelFactory.d.ts',
 	tunnels: 'https://raw.githubusercontent.com/microsoft/vscode/main/src/vscode-dts/vscode.proposed.tunnels.d.ts',
+	viewContainerContext: 'https://raw.githubusercontent.com/microsoft/vscode/main/src/vscode-dts/vscode.proposed.viewContainerContext.d.ts',
```
Comment (jrieken) [inline src/vs/workbench/services/extensions/common/extensionsApiProposals.ts:130]: By our convention this should be called `contribXYZ`

**Review by jrieken — DISMISSED** [review]

File: src/vs/workbench/services/extensions/common/extensionsApiProposals.ts (hunk 1 above)
Comment (thegecko) [reply src/vs/workbench/services/extensions/common/extensionsApiProposals.ts]: Updated reference (in new format) and filename to follow this convention

**Comment by thegecko** [comment]
> It this be viewContainer/... instead of view/container/..?

I've renamed this to `viewContainer/`. The original format was based on items such as this:
https://github.com/thegecko/vscode/blob/9d4274e5599f6ba2c667a54ec0d1d6594d30145b/src/vs/workbench/services/actions/common/menusExtensionPoint.ts#L242

Readme and example updated to reflect the changes above.

**Comment by roblourens** [comment]
Discussed with @jrieken earlier, pushed a change to require that this contribution point only apply to the debug view container, for now. Also, the proposal is mentioning "context menus" but this is only about the title menu, right? I'm renaming the proposal, but otherwise this is ready for Insiders :)

**Review by roblourens — APPROVED** [review]

**Review by Tyriar — APPROVED** [review]

**Pull request merged**